src/cleaner.ts

```
@@ -1,390 +1,535 @@
+			);
+			console.log(
+				"Output may be long, it's possible your console buffer won't show everything.",
+			);
+			console.log(
+				"This command may run for a long time and some results may be lost.",
+			);
+			console.log(
+				"Use -o flag to specify a file such as -o /tmp/repositories.json",
+			);
+			console.log("");
+
+			if (stdin.isTTY && process.env.NODE_ENV !== "test") {
+				await this.promptUser(
+					"Press CTRL+C to interrupt or ENTER to continue...",
+				);
+			} else {
+				console.log(
+					"Non-interactive environment detected. Continuing without prompt.",
+				);
+			}
+		}
+
+		if (startIndex > endIndex) {
+			console.error("Start index is greater than end index");
+			process.exit(1);
+		}
+
+		const totalLength = endIndex - startIndex + 1;
+		const repositoryIds = [...Array(totalLength).keys()].map(
+			(i) => i + startIndex,
+		);
+
+		console.log(
+			`🔭 Requesting container repository IDs [${startIndex}-${endIndex}] with concurrency ${this.concurrency}`,
+		);
+
+		const repositories: RegistryRepositorySchema[] = [];
+
+		// Process repos in batches of size this.concurrency
+		let totalFetched = 0;
+		for (let i = 0; i < repositoryIds.length; i += this.concurrency) {
+			const batch = repositoryIds.slice(i, i + this.concurrency);
+			const batchPromises = batch.map((repositoryId) =>
+				this.getContainerRepositories(repositoryId),
+			);
+
+			console.log(`   Fetching ${batch.length} repositories, ${totalFetched}/${totalLength} done`);
+			const batchResults = await Promise.allSettled(batchPromises);
+			totalFetched += batch.length;
+			for (const result of batchResults) {
+				if (result.status === "fulfilled") {
+					repositories.push(result.value);
+				} // we ignore errors for now (invalid IDs)
+			}
+		}
+
+		if (repositories.length === 0) {
+			console.warn("   No repositories found. Maybe try again with a different ID range?");
+			process.exit(1);
```

Maybe throw an exception here instead ?